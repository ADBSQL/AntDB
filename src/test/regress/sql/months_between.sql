set grammar to oracle;
create table monthsbtwtbl(id int, dt timestamp);
insert into monthsbtwtbl values(1,to_date('2000-3-31','YYYY-MM-DD'));
insert into monthsbtwtbl values(2,to_date('2000-2-29','YYYY-MM-DD'));
select months_between(to_date('2000-3-31','yyyy-mm-dd'),to_date('2000-2-29','yyyy-mm-dd')) from dual;
select id,months_between(dt,to_date('2000-2-29','yyyy-mm-dd')) from monthsbtwtbl;
select months_between(to_date('2000-3-31','yyyy-mm-dd'),to_date('2000-4-30','yyyy-mm-dd')) from dual;
select months_between(to_date('2000-2-29','yyyy-mm-dd'),to_date('1999-2-28','yyyy-mm-dd')) from dual;
select months_between(to_date('1-1-1','yyyy-mm-dd'),to_date('9999-12-1','yyyy-mm-dd')) from dual;
select months_between(null,to_date('9999-12-1','yyyy-mm-dd')) from dual;
select months_between(to_date('2000-2-29','yyyy-mm-dd'),null) from dual;
select months_between(null,null) from dual;
select months_between(to_date('2000-4-30 23:59:59','yyyy-mm-dd hh24:mi:ss'),to_date('2000-4-30 0:0:0','yyyy-mm-dd HH24:MI:SS')) from dual;
drop table monthsbtwtbl;
