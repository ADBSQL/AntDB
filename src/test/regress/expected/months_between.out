set grammar to oracle;
create table monthsbtwtbl(id int, dt timestamp);
insert into monthsbtwtbl values(1,to_date('2000-3-31','YYYY-MM-DD'));
insert into monthsbtwtbl values(2,to_date('2000-2-29','YYYY-MM-DD'));
select months_between(to_date('2000-3-31','yyyy-mm-dd'),to_date('2000-2-29','yyyy-mm-dd')) from dual;
 MONTHS_BETWEEN(TO_DATE('2000-3-31','YYYY-MM-DD'),TO_DATE('2000-2-29','YYYY-MM-DD')) 
-------------------------------------------------------------------------------------
                                                                                   1
(1 row)

select id,months_between(dt,to_date('2000-2-29','yyyy-mm-dd')) from monthsbtwtbl;
 ID | MONTHS_BETWEEN(DT,TO_DATE('2000-2-29','YYYY-MM-DD')) 
----+------------------------------------------------------
  1 |                                                    1
  2 |                                                    0
(2 rows)

select months_between(to_date('2000-3-31','yyyy-mm-dd'),to_date('2000-4-30','yyyy-mm-dd')) from dual;
 MONTHS_BETWEEN(TO_DATE('2000-3-31','YYYY-MM-DD'),TO_DATE('2000-4-30','YYYY-MM-DD')) 
-------------------------------------------------------------------------------------
                                                                                  -1
(1 row)

select months_between(to_date('2000-2-29','yyyy-mm-dd'),to_date('1999-2-28','yyyy-mm-dd')) from dual;
 MONTHS_BETWEEN(TO_DATE('2000-2-29','YYYY-MM-DD'),TO_DATE('1999-2-28','YYYY-MM-DD')) 
-------------------------------------------------------------------------------------
                                                                                  12
(1 row)

select months_between(to_date('1-1-1','yyyy-mm-dd'),to_date('9999-12-1','yyyy-mm-dd')) from dual;
 MONTHS_BETWEEN(TO_DATE('1-1-1','YYYY-MM-DD'),TO_DATE('9999-12-1','YYYY-MM-DD')) 
---------------------------------------------------------------------------------
                                                                         -119987
(1 row)

select months_between(null,to_date('9999-12-1','yyyy-mm-dd')) from dual;
 MONTHS_BETWEEN(NULL,TO_DATE('9999-12-1','YYYY-MM-DD')) 
--------------------------------------------------------
                                                       
(1 row)

select months_between(to_date('2000-2-29','yyyy-mm-dd'),null) from dual;
 MONTHS_BETWEEN(TO_DATE('2000-2-29','YYYY-MM-DD'),NULL) 
--------------------------------------------------------
                                                       
(1 row)

select months_between(null,null) from dual;
 MONTHS_BETWEEN(NULL,NULL) 
---------------------------
                          
(1 row)

select months_between(to_date('2000-4-30 23:59:59','yyyy-mm-dd hh24:mi:ss'),to_date('2000-4-30 0:0:0','yyyy-mm-dd HH24:MI:SS')) from dual;
 MONTHS_BETWEEN(TO_DATE('2000-4-30 23:59:59','YYYY-MM-DD HH24:MI:SS'),TO_DATE('2000-4-30 0:0:0','YYYY-MM-DD HH24:MI:SS')) 
--------------------------------------------------------------------------------------------------------------------------
                                                                                                                        0
(1 row)

drop table monthsbtwtbl;
